server/application-server/src/main/java/de/tum/in/www1/hephaestus/gitprovider/team/Team.java

```diff
@@ -0,0 +1,82 @@
+package de.tum.in.www1.hephaestus.gitprovider.team;
+
+import jakarta.persistence.Table;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.lang.NonNull;
+import de.tum.in.www1.hephaestus.gitprovider.label.Label;
+import de.tum.in.www1.hephaestus.gitprovider.repository.Repository;
+import de.tum.in.www1.hephaestus.gitprovider.user.User;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.ManyToMany;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Table(name = "team")
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString(callSuper = true)
+public class Team {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column
+    private String name;
+
+    @NonNull
+    private String color;
+
+    @ManyToMany
+    @JoinTable(name = "team_repositories", joinColumns = @JoinColumn(name = "team_id"), inverseJoinColumns = @JoinColumn(name = "repository_id"))
```

Comment:
So my idea to solve this without the need for a new table would be the following:
- since labels are already connected to exactly one repository, we can reuse that connection instead
- if we add a whole repository to a team, it means we include it independent of labels ("all labels")

I do understand the confusion and probably should update the UI accordingly to clarify my idea.

Reply:
Ah I see, yeah this works for me :) That's a good solution! But you need to reflect it in some way in the UI. 
Maybe you can group them with the repository visually?

Reply:
I reworked the UI in 885a37c now. Now we got a visual grouping of labels + repositories. Let me know if this layout is still missing something. 
![image](https://github.com/user-attachments/assets/1528416f-bdbb-44a7-9267-c9e67417c2e5)

Reply:
I think that works. Thank you!